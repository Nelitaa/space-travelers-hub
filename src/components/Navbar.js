import { Link, NavLink } from 'react-router-dom';
import logo from '../images/planet.png';

const Navbar = () => {
  const links = [
    {
      id: 1,
      title: 'My Profile',
      to: '/profile',
    },
  ];
  return (
    <header className="header-container">
      <Link to="/" className="brand-container">
        <img src={logo} alt="Space Travel Logo" className="logo" />
        <p className="logo-text">Space Travelers&apos; Hub</p>
      </Link>
      <nav className="nav-container">
        <ul className="links-list">
          {links.length > 0
          && links.map((link) => (
            <li className="link-item" key={link.id}>
              <NavLink to={link.to}>
                {link.title}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default Navbar;
