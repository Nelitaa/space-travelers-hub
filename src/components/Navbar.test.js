import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Navbar from './Navbar';

test('renders Navbar component', () => {
  const { asFragment } = render(<Router><Navbar /></Router>);
  expect(asFragment()).toMatchSnapshot();
});

test('render Navbar component with links', () => {
  const { getByText } = render(<Router><Navbar /></Router>);
  const rocketsLink = getByText(/Rockets/i);
  const missionsLink = getByText(/Missions/i);
  const profileLink = getByText(/My Profile/i);
  expect(rocketsLink).toHaveAttribute('href', '/');
  expect(missionsLink).toHaveAttribute('href', '/missions');
  expect(profileLink).toHaveAttribute('href', '/profile');
});
