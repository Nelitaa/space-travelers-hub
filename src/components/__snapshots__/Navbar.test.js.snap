// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar component 1`] = `
<DocumentFragment>
  <header
    class="header-container"
  >
    <a
      class="brand-container"
      href="/"
    >
      <img
        alt="Space Travel Logo"
        class="logo"
        src="planet.png"
      />
      <p
        class="logo-text"
      >
        Space Travelers' Hub
      </p>
    </a>
    <nav
      class="nav-container"
    >
      <ul
        class="links-list"
      >
        <li
          class="link-item link-item-active"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li
          class="link-item "
        >
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="link-item "
        >
          <a
            class=""
            href="/profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
